### ISSUE TITLE
Eloquent Builder array conditions need `@psalm-taint-escape sql`

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#685]
Eloquent's `where()`, `firstWhere()`, `find()`, `findOrFail()` etc. use PDO parameterized queries when passed array conditions — values are never interpolated into raw SQL. But taint analysis tracks user input flowing into these methods as `TaintedSql`.

Example from pixelfed:
```php
$pid = $request->input('pid'); // tainted
$sid = $request->input('sid'); // tainted

DirectMessage::whereToId($pid)->whereFromId($sid)->...  // false positive TaintedSql
GroupLike::where(['group_id' => $gid, 'comment_id' => $sid])->first();  // false positive
```

Fix: add `@psalm-taint-escape sql` to the safe Builder methods in stubs. Methods that must NOT get the escape: `whereRaw`, `selectRaw`, `havingRaw`, `orderByRaw`, `DB::statement`, `DB::unprepared`.

Observed in both pixelfed (4 issues) and monica (1 issue).
